nemo/collections/asr/models/ctc_models.py

```diff
@@ -332,24 +348,49 @@ def output_types(self) -> Optional[Dict[str, NeuralType]]:
             "greedy_predictions": NeuralType(('B', 'T'), LabelsType()),
         }
 
-    @typecheck()
-    def forward(self, input_signal, input_signal_length):
-        processed_signal, processed_signal_len = self.preprocessor(
-            input_signal=input_signal, length=input_signal_length,
-        )
+    def processed_batch(self, batch):
+        need_processing = True
         # Spec augment is not applied during evaluation/testing
-        if self.spec_augmentation is not None and self.training:
+        need_augmenting = self.spec_augmentation is not None and self.training
+        if len(batch) == 1:  # DALI case
```

Comment:
This will break when we perform batch size 1 evaluation with current data loaders (required for score reporting in publications). Please provide some additional check to ensure Dali pipeline is being run here.

Reply:
I believe that `len(batch)` is not the batch size, but it is the size of the array containing the 4 outputs, so it will always be 4 for the non-dali case. That being said, I also feel this check is not good enough. I'll try to come up with something clearer.

Reply:
You're right, this is the unpacked tuples from the dataloader, so 4 for the standard ones and 1 for Dali. Making it explicit somehow would possibly be better.